src/frontend/src/binder/set_expr.rs

```diff
@@ -160,6 +160,7 @@ impl Binder {
                         // select a from t2 union all select b from t2 order by a+1; should throw an
                         // error.
                         self.context = BindContext::default();
+                        self.context.cte_to_relation = new_context.cte_to_relation;
```

Comment:
since CTE can only be defined at the top level, I suppose I can just retain the `cte_to_relation` without concern?